src/util/triplea/MapXMLCreator/MapXMLCreator.java

```diff
@@ -0,0 +1,839 @@
+						break;
+					case 15:
+						final String[] settings = GameSettingsPanel.s_allSettings.split(", ");
+						Arrays.sort(settings);
+						StringBuilder sb = new StringBuilder();
+						sb.append("<html>Here is a list of all the available game settings:<table border=0><tr><td><table border=0>");
+						int setting_nr = 1;
+						char last_first_letter = 'z';
+						final int third_size_setting = (settings.length + 2) / 3;
+						for (final String setting : settings)
+						{
+							char first_letter = setting.charAt(0);
+							if (first_letter == last_first_letter)
+								sb.append("<tr><td>" + setting + "</td></tr>");
+							else
+								sb.append("<tr><td><big>" + first_letter + "</big>" + setting.substring(1) + "</td></tr>");
+							last_first_letter = first_letter;
+							if (setting_nr % third_size_setting == 0)
+								sb.append("</table></td><td><table border=0>");
+							++setting_nr;
+						}
+						sb.append("</table></td></tr></table></html>");
+						showInfoMessage(
+sb.toString(), "List Of Available Game Settings Choices");
+						break;
+					case 16:
+						showInfoMessage(
+									"The map notes can be entered in the textbox. You can enter plain text or html code. Then just click the 'Save Map To File' button, and save the file in the map's 'games' folder. (Create the 'games' folder if not already created)",
+									"Help For Current Step");
+						break;
+				}
+			}
+		});
+		southRightPanel.add(m_bAvailableChoices);
+	}
+	
+	private void showInfoMessage(final String message, final String title) {
+		JOptionPane.showMessageDialog(null,message,title,JOptionPane.INFORMATION_MESSAGE);
+	}
+	
+	private void setupMainPanel(final JPanel panel)
+	{
+		m_mainPanel.removeAll();
+		m_mainPanel.setLayout(new BorderLayout());
+		panel.setBorder(new MatteBorder(1, 0, 1, 1, Color.DARK_GRAY));
+		m_mainPanel.add(panel,BorderLayout.CENTER);
+		setWidgetActivation();
+	}
+	
+	public void setWidgetActivation()
+	{
+		if (!SwingUtilities.isEventDispatchThread())
+		{
+			SwingUtilities.invokeLater(new Runnable()
+			{
+				public void run()
+				{
+					setWidgetActivation();
+				}
+			});
+			return;
+		}
+		m_mainPanel.revalidate();
+		m_mainPanel.repaint();
+		this.validate();
+		this.repaint();
+	}
+	
+	private void goToStep(final int step) {
+		JLabel stepLabel;
+		if (m_currentStep >= 1) {
+			stepLabel = m_stepList.get(m_currentStep-1);
+			stepLabel.setFont(new Font("Tahoma", Font.PLAIN, 13));
+			stepLabel.repaint();
+		}
+		
+		if (step <= m_stepList.size())
+		{
+			m_currentStep = step;
+			m_highest_step = Math.max(m_highest_step, step);
+			layoutStepActionPanel();
+			
+			stepLabel = m_stepList.get(step-1);

```

Comment:
If the above condition is true, m_currentStep >= 1, then we will overwrite 'stepLabel' and do an additional repaint. Should the two if statements be an if/else block instead?